src/acl/external/AD_group/ext_ad_group_acl.cc

```diff
@@ -389,17 +390,18 @@ add_User_Group(wchar_t * Group)
     return 1;
 }
 
-/* returns 0 on match, -1 if no match */
-static int
-wccmparray(const wchar_t * str, const wchar_t ** array)
+/* returns true on match, false if no match */
+/* TODO: convert to std::containers */
+static bool
+wccmparray(const wchar_t * str, wchar_t ** array)
 {
     while (*array) {
```

Comment:
Since this PR changes this function API, I recommend changing the second parameter type to remove one pointer dereference so that callers know that their (old) array pointer (to pointer) cannot be nil. In other words, so that existing callers would have to write `wccmparray(item, *container_ptr_ptr)` instead of the more dangerous `wccmparray(item, container_ptr_ptr)` in the current code. Alternatively, add a check for nil `array`.

The suggestion below is not sufficient to make the code compile after the above changes, but it illustrates the point.

```suggestion
    while (array) {
```

Reply:
added the explicit test for non-null array at the beginning of the function